x-pack/plugins/ml/public/application/jobs/components/custom_url_editor/utils.js

```diff
@@ -289,64 +286,53 @@ export function getTestUrl(job, customUrl) {
     sort: [{ record_score: { order: 'desc' } }],
   };
 
-  return new Promise((resolve, reject) => {
-    ml.results
-      .anomalySearch(
-        {
-          body,
-        },
-        [job.job_id]
-      )
-      .then((resp) => {
-        if (resp.hits.total.value > 0) {
-          const record = resp.hits.hits[0]._source;
-          testUrl = replaceTokensInUrlValue(customUrl, bucketSpanSecs, record, 'timestamp');
-          resolve(testUrl);
-        } else {
-          // No anomalies yet for this job, so do a preview of the search
-          // configured in the job datafeed to obtain sample docs.
-          mlJobService.searchPreview(job).then((response) => {
-            let testDoc;
-            const docTimeFieldName = job.data_description.time_field;
-
-            // Handle datafeeds which use aggregations or documents.
-            if (response.aggregations) {
-              // Create a dummy object which contains the fields necessary to build the URL.
-              const firstBucket = response.aggregations.buckets.buckets[0];
-              testDoc = {
-                [docTimeFieldName]: firstBucket.key,
-              };
-
-              // Look for bucket aggregations which match the tokens in the URL.
-              urlValue.replace(/\$([^?&$\'"]{1,40})\$/g, (match, name) => {
-                if (name !== 'earliest' && name !== 'latest' && firstBucket[name] !== undefined) {
-                  const tokenBuckets = firstBucket[name];
-                  if (tokenBuckets.buckets) {
-                    testDoc[name] = tokenBuckets.buckets[0].key;
-                  }
-                }
-              });
-            } else {
-              if (response.hits.total.value > 0) {
-                testDoc = getProcessedFields(response.hits.hits[0].fields);
-              }
-            }
-
-            if (testDoc !== undefined) {
-              testUrl = replaceTokensInUrlValue(
-                customUrl,
-                bucketSpanSecs,
-                testDoc,
-                docTimeFieldName
-              );
-            }
+  let resp;
+  try {
+    resp = await ml.results.anomalySearch(
+      {
+        body,
+      },
+      [job.job_id]
+    );
+  } catch (error) {
+    // search may fail if the job doesn't already exist
+  }
```

Comment:
do we want to check for a particular error code/message here? otherwise it might silently fail unintentionally

Reply:
Failures here will cause failures later on which are not caught and so will be displayed to the user.
I also am not sure what errors we'd need to check for and accept.
This is also a feature that isn't used that much. It has been broken for a long time and no one has noticed.